packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -102,19 +138,23 @@ export default class MetricService {
    */
   constructor(
     logger: Logger,
-    sdkClient: SDKClient,
+    hapiService: HAPIService,
     mirrorNodeClient: MirrorNodeClient,
     hbarLimiter: HbarLimit,
     register: Registry,
     eventEmitter: EventEmitter,
   ) {
     this.logger = logger;
-    this.sdkClient = sdkClient;
+    this.hapiService = hapiService;
+    this.sdkClient = hapiService.getSDKClient();
```

Comment:
But do we really need this, not sure we need the metric service to depend on the sdkClient. The sdkClient is used here to get details from the transaction record in the captureTransactionMetrics. However, arent they passed when emitting the event in the sdkClient?

Reply:
This is a really good point! Perhaps we can passed the `transactionRecordMetrics` in when emitting the event along with other arguments. But I guess it's gonna be out of the scope for this PR we can create a ticket for it. Great point!